### ISSUE TITLE
Admin Feature: Show which users have flagged something

### ISSUE THREAD

**Issue by sizzlemctwizzle** [OpenUserJs/OpenUserJS.org#643]
Related to #642. Admins need a way to see who flagged something to remove users that flag things for no conceivable reason.

**Comment by TimidScript** [comment]
Continuation of discussion #640 (relevant: #641).  I choose to reply here as it is more relevant.

May I suggest force them to provide a flag feedback comment and with a warning on abuse. Abuse can result in the removal of flagging ability or on extreme cases, removal of user (though I think it is very harsh option). If people must provide a reason, it will cause less abuse and less flagging. 

Also waiting until script has positive flag count is risking it, and relying on admin to catch those that fall through the crack is way too much considering the amount of scripts. 

So an open forum that allows users to monitor open flagged feedback. When the flag thread is open are able to access it and even provide input. If the scripts flag thread is open, and a new user tries to flag the script he will be pointed to the open thread and will be asked to add his issue at the end. 

If it is an flagged erroneously or is resolved, the admin deflags the thread and the post is then hidden from non-admin. If another flag is raised after that, a new flag thread is created for the script.

**Comment by Martii** [comment]
As I've stated to you directly before TimidScript we don't need a flame war. Flagging will eventually have a reason once #262 gets the the `get`s to a `post` and we can place reasons in... much like we have reasons for removal on the admin+ side. Depending on the abuse of the flagging reasons e.g. someone trying to force their username in... we **might** consider making it available to lower privileged owners of each script... however script discussions are the place where someone should be discussing issues to improve things in a script and downvoting is a way of "disapproval" without involving moderation techniques.

**Comment by sizzlemctwizzle** [comment]
I agree with @Martii. Issues exist for discussion on how to resolve problems with a script (and request features and whatnot). Flags are to report malicious activity, which really needs no discussion on the part of ordinary users. You will be asked to give a reason while flagging once that feature is finally implemented.

> Also waiting until script has positive flag count is risking it, and relying on admin to catch those that fall through the crack is way too much considering the amount of scripts.

Honestly the case of a very popular script suddenly turning malicious is a rare edge case. Admins using #641 would catch these exceptions with ease.

**Comment by Martii** [comment]
Isolating this to just the admins is nice... but there was a comment somewhere about letting individual flaggers see what they have flagged in case they change their mind e.g. my thoughts are showing the list/link on each user accounts homepage to the owners and also make it available to admins+... a separate listing of each script is what is being mentioned here but I think the root'ing should be on each user first.

**Comment by Martii** [comment]
So now that the Admins and up can view this on production... any final UI styling changes before closing?

Reasons will be added after #641 is determined... so please go visit there too.

**Comment by Martii** [comment]
Just a misc note for current Moderators... on the script home page to tell if a script is critically flagged look at the rating bar and if it's **all** red with no grey then it's critically flagged... and of course it will show up in the Moderator tools.

**Comment by Martii** [comment]
Only thing I can possibly think of at this time is to add the current "role" to the User button tooltip but that's not as important so closing as fixed... #641 will enable a more finite list when completed but this structure. Opening new issue for "reason".

**Comment by Martii** [comment]
Whoops have one view bug of not maintaining the library QSP... fixing momentarily.
